sphire/protocols/protocol_cryolo_napari_tomo_picking.py

```diff
@@ -65,29 +65,40 @@ def prepareDataStep(self):
         convert.convertMicrographs(tomoList, self._getExtraPath())
 
     def runCoordinatePickingStep(self):
-        # Getting the first tomogram to check if the .cbox file exist
-        tomogram = self.getInputTomos().getFirstItem()
-        filePath = os.path.join(self._getExtraPath(),
-                                convert.getMicFn(tomogram, "cbox"))
-        creationOldTime = None
-        if os.path.exists(filePath):
-            creationOldTime = time.ctime(os.path.getctime(filePath))
+        """Run napari-boxmanager"""
+
+        fileDict = {}
+        # Finding the coordinate file per tomogram
+        for tomogram in self.getInputTomos().iterItems():
+            filePath = os.path.join(self._getExtraPath(),  convert.getMicFn(tomogram, "cbox"))
+            if not os.path.exists(filePath):
+                filePath = os.path.join(self._getExtraPath(), convert.getMicFn(tomogram, "coords"))
+
+            if os.path.exists(filePath):
+                creationOldTime = time.ctime(os.path.getctime(filePath))
+                fileDict[filePath] = creationOldTime
 
         view = SphireGenericView(None,
                                  [tomo.clone() for tomo in self.getInputTomos()],
                                  self._getExtraPath(), isInteractive=True)
         view.show()
 
-        if os.path.exists(filePath):
-            if creationOldTime is not None:
+        for tomogram in self.getInputTomos().iterItems():
+            filePath = os.path.join(self._getExtraPath(), convert.getMicFn(tomogram,  "cbox"))
```

Comment:
convert.getMicFn(tomogram,  "cbox") is the worst :) Passing a whole tomogram object when you only need the filename to convert..